west.yml

```diff
@@ -12,6 +12,24 @@ manifest:
     - name: zephyrproject
       url-base: https://github.com/zephyrproject-rtos
 
+  # When upgrading projects here please run git log --oneline in the
+  # project and if not too long then include the output in your commit
+  # message. See examples of good commit messages at
+  # https://github.com/thesofproject/sof/commits/4e1d3ba61abd3/rimage
+  # (git log 4e1d3ba61abd3 -- rimage/)
+
+  # WARNING: for backwards compatibility with non-Zephyr platforms,
+  # rimage and toml99 are STILL git submodules! Every time you change
+  # their `revision` revision here you MUST keep their git submodules in
+  # sync. This is as simple as `git add rimage` before committing (+
+  # `git commit --amend` if you forgot and already committed this
+  # west manifest file).
+
+  # When a submodule is out of sync then both `git status` and `git
+  # describe` notice and a "-dirty" suffix appears in the SOF version.
+  # It is generally very bad practice to git push without looking
+  # at "git status"; even without submodules.
+
```

Comment:
Given we will need frequent updates to rimage (like https://github.com/thesofproject/sof/pull/6385 today), the above seems just way too complicated. Can we use a different directory for either the submodule (renaming it) or the west module?  This probably needs to be done in steps to remove the hardcodes for rimage path, before the switch, but I don't see we can survive with this duplicated rimage copies setup in the long term.

Not blocking the PR because of this, as long as we use the same directory name, this is needed yes or yes.

Reply:
Yes, the best would be to move rimage one level-up in west.yml and leave the rimage submodule in place. west would then stop fooling git submodules. This would help de-hardcoding the `sof` name of the sof clone too.

BTW is suspiciously quiet about nesting git repos. It does not explicitly forbid it but it never shows any nested git repo in any example anywhere in the documentation.

HOWEVER, moving rimage would require large, disruptive and backwards-incompatible changes in the build system and in CI, which is why it was not done yet. I mean much more complicated and much more time-consuming changes than keeping the submodule in sync. Good request to add the backlog though.

Reply:
> Given we will need frequent updates to rimage (like https://github.com/thesofproject/sof/pull/6385 today),

BTW rimage updates were not really frequent in the past. They are more frequent now because MTL is brand new but I would expect this to ease off soon. As long as the firmware boots who cares about rimage?